osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -168,6 +169,11 @@ protected virtual void CreateNestedHitObjects()
         /// </summary>
         [NotNull]
         protected virtual HitWindows CreateHitWindows() => new HitWindows();
+
+        /// <summary>
+        /// The result this <see cref="HitObject"/> was judged with. Used internally for rewinding within <see cref="DrawableHitObject"/>.
+        /// </summary>
+        internal JudgementResult Result;
```

Comment:
Hmm, sure about this one? This is the first time we are storing gameplay-related state inside the hitobject model.

Reply:
I mean the result has to get stored *somewhere* if the drawable objects are to be recycled, and this seems to be the immediate and simplest solution.

Reply:
I get that, but it also breaks the pattern we have been going for until now (`HitObject`s are never mutable by gameplay).

Reply:
I'll try to move it into the lifetime entries in my pooling implementation again. I tried that originally but not sure why it didn't work out.

Reply:
Honestly not sure it's so bad where it is, but would definitely help for peace of mind. I can only imagine we'll run into issues down the track (editor, serialisation, etc.) if it's in `HitObject`.

Also doesn't set a great precedent (even though it's `internal`).

Reply:
Right, I remember what the issue is. Storing it inside the hitobject lifetime entries works for pooling top-level hitobjects, but doesn't work for pooling nested hitobjects. The alternative to this, I think, is to store a `Dictionary<HitObject, HitObjectLifetimeEntry>` in `DrawableRuleset`, and then either retrieve or construct dummy entries on-the-fly as drawable representations are requested.

I do agree that having this stored in `HitObject` doesn't set a good precedent.

Reply:
Tested and my idea does seem to work: https://github.com/smoogipoo/osu/commit/a6174a0e3199f6a0306aaddb2200fadedcc97de5

I've removed the storage in hitobject from this PR.